src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -697,6 +698,23 @@ namespace OpenMS
       return *this;
     }
 
+    /// get the file path to the first MS run
+    StringList getPrimaryMSRunPath() const
+    {
+      StringList ms_run_paths;
+      std::vector<SourceFile> sfs(this->getSourceFiles());
+      for (std::vector<SourceFile>::const_iterator it = sfs.begin(); it != sfs.end(); ++it)
+      {
+        // assemble a single location string from the URI (path to file) and file name
+        String ms_run_location = it->getPathToFile() + "/" + it->getNameOfFile();
+        if (!ms_run_location.empty())
```

Comment:
this cannot be empty, since we at least add "/".

How about a SourceFile::isEmpty() function?!

btw: if someone would just set the SourceFile:filename and leave the SourceFile:path empty, we would get an invalid path here, right?! (unless getPathToFile() would be ".".